services/streammanager/kafka/kafkamanager.go

```diff
@@ -533,41 +544,27 @@ func (p *ProducerManager) Produce(jsonData json.RawMessage, destConfig interface
 		// return 400 if producer is invalid
 		return 400, "Could not create producer", "Could not create producer"
 	}
-
 	start := now()
 	defer func() { kafkaStats.produceTime.SendTiming(since(start)) }()
 
-	conf := configuration{}
-	jsonConfig, err := json.Marshal(destConfig)
-	if err != nil {
-		return makeErrorResponse(err) // returning 500 for retrying, in case of bad configuration
-	}
-	err = json.Unmarshal(jsonConfig, &conf)
-	if err != nil {
-		return makeErrorResponse(err) // returning 500 for retrying, in case of bad configuration
-	}
-
-	if conf.Topic == "" {
-		return makeErrorResponse(fmt.Errorf("invalid destination configuration: no topic"))
-	}
-
```

Comment:
You're removing these sanity checks from here but where are we checking if `destConfig` is a valid JSON now? If we're planning to drop this check please rename `destConfig` to `_` (function parameter).

Reply:
The configs are used now, added the sanity test cases as well